### DISCUSSION TITLE
Adobe Animiate AAB Creation

### DISCUSSION THREAD

**Discussion by gmwtrgs** [airsdk/Adobe-Runtime-Support#1137] [Q&A]
I've just started to try to create my first AAB file for posting to Google Play and having no end of problems!

I know there is a thread about this but does anyone have an idiots guide they could post?

I have tried altering the ADT.CFG file as shown below:

# To override the creation of an Android APK file, and to instead create an Android App Bundle, the below
# flag can be uncommented. Note that the bundle may end up with an "apk" file extension
CreateAndroidAppBundle=true

# When creating an Android App Bundle, this flag can be used to output the Android Studio project
# files in a subfolder of your build directory (rather than using a temporary location that gets
# cleaned up on exit).
#KeepAndroidStudioOutput=true

# For convenience, rather than using "-platformsdk" on the command line, these can be set up here:
#AndroidPlatformSDK=path_to_sdk
AndroidPlatformSDK=C:\\Users\\Gee\\AppData\\Local\\Android\\Sdk
#iOSPlatformSDK=path_to_sdk

# JAVA_HOME is usually picked up from an environment variable but this can override it:
JAVA_HOME=c:\\Program Files\\Android\\Android Studio\\jre

I've also installed Android Studio (which other people have said needs to be done) but every time I try to publish from with Animate it give me a raft of errors which I don't understand.

Any help would be gratefully received.

![2021-09-08_17-55-25](https://user-images.githubusercontent.com/52746140/132553274-d208fa2f-105f-4ff4-b184-19d370307dd8.jpg)

**Comment by ajwfrost** [comment]
Well interestingly, I am surprised it gets as far as it does if those paths are in your config file -> for Windows, paths should be using either a double-backslash, or a forward-slash. e.g. `AndroidPlatformSDK=C:/Users/Gee/AppData/Local/Android/Sdk`

The actual error that's being output there is at the very bottom of the dialog screenshot:
`Installed Build Tools revision 31.0.0 is corrupted. Remove and install again using the SDK Manager.`

Someone else had reported this I think but I'm not sure there's a solution if using build tools 31. Do you have an older version available? If you go into `C:\Users\Gee\AppData\Local\Android\Sdk\build-tools` please check if you have a 30.0.x folder... if you don't, you can download one from Android Studio or the SDK Manager (un-tick the "Hide Obsolete Packages".

Then to prevent ADT from picking up the 31.0.0 folder, try changing it to 'hidden' via the file properties.

In the meantime, we'll look at what might be up with 31.0.0, it may just be that there's incompatibility between different versions of Gradle and the Gradle Android Plug-in vs the SDK build tools...

thanks

**Reply by GeneralVimes** [reply]
I specifically mentioned how to fix this in my tutorial: https://www.reddit.com/r/as3/comments/owag5h/building_aab_with_flash_develop_working_experience/

Here: 
In the SDK manager we can see the folder, where Android SDK will be downloaded [A]. You'll need it on the next step. By default Android Studio wants to install SDK for Android 12 (API 31). But as we are targeting API 30, we must unselect Android 12 SDK in "SDK Platforms" (B), and select Android 11 SDK. And, which important, too, in the tab "SDK Tools" (C) you should select the checkbox "Show Package details"(D), then unselect "31" in "Android SDK Build Tools" and select "30" there (E).

https://imgur.com/a/rshYFZI
